src/sidebar/components/MarkdownEditor.tsx

```diff
@@ -202,7 +202,7 @@ type TextAreaProps = {
   classes?: string;
   containerRef?: Ref<HTMLTextAreaElement>;
   mentionsEnabled: boolean;
-  usersForMentions: UserItem[];
+  usersForMentions: UserItem[] | 'loading';
```

Comment:
Modeling a sum type like this crates a hazard because some operations are valid for both strings and arrays, but would lead to incorrect results if the value is a string. `usersForMentions.length` will work for both (since strings and arrays both have a length property), but doesn't make sense for the string case.

Instead I would suggest using `{ state: 'loading' } | { state: 'loaded', users: UserItem[] }`. Alternatively you could use `null` to represent the state that users are being loaded. The downside is that there is no way of extending this to represent an error later.

Reply:
Yeah, this is a good point. I did in fact face this issue, and this comment here is a side effect of that https://github.com/hypothesis/client/pull/6862#discussion_r1983658074, but I think that was not the right solution, just addressing a symptom.

Your proposal makes way more sense.